google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ClientSideStatementValueConverters.java

```diff
@@ -85,6 +88,43 @@ public Boolean convert(String value) {
     }
   }
 
+  /** Converter from string to {@link Boolean} */
+  static class PgBooleanConverter implements ClientSideStatementValueConverter<Boolean> {
+
+    public PgBooleanConverter(String allowedValues) {}
+
+    @Override
+    public Class<Boolean> getParameterClass() {
+      return Boolean.class;
+    }
+
+    @Override
+    public Boolean convert(String value) {
+      if (value == null) {
+        return null;
+      }
+      if (value.length() > 1
+          && ((value.startsWith("'") && value.endsWith("'"))
+              || (value.startsWith("\"") && value.endsWith("\"")))) {
+        value = value.substring(1, value.length() - 1);
+      }
+      if ("true".equalsIgnoreCase(value)
```

Comment:
I think we should also trim the string to consider the cases like `'  true  '`

Reply:
(See also above). PostgreSQL does not consider that to be a valid boolean value. When the value is quoted with either single or double quotes, it may not contain any spaces.